### PR TITLE
use rpam2, cleanup

### PR DESCRIPTION
**Pull request by devkral**
* Switch application to use rpam2
* Use listenv feature of rpam2 to get email and other possible variables. This works only if the pam environment is populated; one other project of mine (anyway it is very easy to populate and could integrate remote data in contrast to Etc)
* some code cleanup
* update dependencies to safe defaults (after last omniauth security fixes)

### CODE REVIEW

**Comment by nickcharlton** [comment]
Hi @devkral,

Thank you for opening this! This is broadly looking pretty good, but would you be willing to open a series of topical PRs for the changes above?

For a long time, I've wanted to add a test suite to this (it's pretty stable, but quite a manual process for testing). Whilst you were doing this, did you come across any thoughts about how we might be able to do that?

**Comment by devkral** [comment]
tests can be done with specs in ruby (rspec).

I can split the pull-request later.
Anyway there is a good tool for extracting single commits: git cherry-pick.

**Comment by devkral** [comment]
split complete. I rebased the commits out so there is no conflict

File: lib/omniauth/strategies/pam.rb [hunk 1]
```diff
@@ -38,21 +38,20 @@ def callback_phase
...
-        info[:email] = "#{uid}@#{options[:email_domain]}" if options.has_key?(:email_domain)
-        info.merge!(parse_gecos || {})
-      end
+        info = { name: uid }
 
-      private
+        list_env = Rpam2.listenv(options[:service],
+                                 uid,
+                                 request["password"],
+                                 false)
```
Comment (nickcharlton) [inline lib/omniauth/strategies/pam.rb]: I'm finding that this is always returning an empty hash on a standard Ubuntu system. Previously, we were extracting the user's name from `gecos` (where, if they'd set it, could also have provided `location`, `phone`, `home_phone`, `description`).

What's different here with `Rpam2`? Do other things need to be installed on the system for this to work correctly?

Comment (devkral) [reply lib/omniauth/strategies/pam.rb]: yes, I forgot to document because the tool was not built then:
https://github.com/devkral/pam_email_extractor

Comment (devkral) [reply lib/omniauth/strategies/pam.rb]: but you can use everything which injects "email" in your pam environment

**Comment by devkral** [comment]
did I mention that I hate linters :)
Everything should be fine now.

File: spec/omniauth/strategies/pam_spec.rb [hunk 2]
```diff
@@ -63,17 +76,12 @@ def invalid_credentials
   end
 
   def mock_rpam(username:, password:, opts:)
-    allow(Rpam).to receive(:auth).with(username, password, opts)
+    allow(Rpam2).to receive(:auth).with(opts[:service], username, password)
```
Comment (nickcharlton) [inline spec/omniauth/strategies/pam_spec.rb]: If we're using a fake with `Rpam2` now, I wonder if we need the mocking too. What do you think?

Comment (devkral) [reply spec/omniauth/strategies/pam_spec.rb]: sry for the late answer: no we don't need it anymore.

**Pull request merged**